### ISSUE TITLE
blobService.createBlockBlobFromBrowserFile is not a function

### ISSUE THREAD

**Issue by meds** [Azure/azure-storage-node#404]
Running the following:

```
 var blobService = azure.createBlobService("key")
blobService.createBlockBlobFromBrowserFile(...)
```

Browser outputs error:

> blobService.createBlockBlobFromBrowserFile is not a function
>     at XMLHttpRequest.xhr.onload 

Other blobService functions behave as expected..

**Comment by XiaoningLiu** [comment]
Hi @meds , what's the browser environment and library version?

**Comment by meds** [comment]
Hey @XiaoningLiu I'm using Chrome with the latest version of the azure-storage npm plugin.

**Comment by XiaoningLiu** [comment]
@meds Please use JavaScript client library for browsers directly, the API is not available on Node.js library.

https://github.com/Azure/azure-storage-node#azure-storage-javascript-client-library-for-browsers

**Comment by vinjiang** [comment]
Feel free to let us know if further assistance required.

**Comment by tonytontian** [comment]
why this API is still in the document, its so confusing.

**Comment by XiaoningLiu** [comment]
@tonytontian All browser only APIs are marked with "(Only available in the JavaScript Client Library for Browsers)" in document and only available for  https://github.com/Azure/azure-storage-node#azure-storage-javascript-client-library-for-browsers Sorry for the inconvenience.

**Comment by sposaro** [comment]
@XiaoningLiu I'm also very confused about how to use this API (and I'm a MSFT). I'm also seeing the same error. Can you please see what is wrong with my react app?

`import React from 'react';
var storage = require('azure-storage');

const CONNECTION_STRING = "MYCONNECTIONSTRING";
const BlockBlobContainerName = "MYCONTAINERNAME";
const BlobName = "MYBLOBNAME";

class NumberUploader extends React.Component {

   render() {
      return (
         <div className="App">

         <input type="file" onChange={ (e) => this.buttonClick(e.target.files[0]) } />

         </div>
      );
   }

   buttonClick(myFile) {
      var blobService = storage.createBlobService(CONNECTION_STRING);

      console.log('1. Creating Container');
      blobService.createContainerIfNotExists(BlockBlobContainerName, function (error) {
         if (error) {
            console.log('error creating container');
         }
      });

      console.log('2. Uploading BlockBlob');

      blobService.createBlockBlobFromBrowserFile(BlockBlobContainerName, BlobName, myFile, function (error, result, response) {
         if (error) {
            alert('Upload failed, open browser console for more detailed info.');
            console.log(error);
         } else {
            setTimeout(function () { // Prevent alert from stopping UI progress update
               alert('Upload successfully!');
            }, 1000);
         }
      });
   }
};
export default NumberUploader;`

**Comment by XiaoningLiu** [comment]
@sposaro 

Here are some differences between storage JS v2 (in this repo) and V10 SDK regarding browser scenarios support:

V10 support browser scenarios with npm package and classic single JS bundle file; V2 only supports browser usage with classic single JS bundle file.

V10 doesn't support SharedKeyCredential in browsers, while V2 supports.

So, if you are building a React Web APP and importing storage SDK using npm dependency. Please use V10. Also please don't use account name and key in browsers, because it's not safe.

**Comment by SogoGolf** [comment]
i thought one of MSFT's goals is to be where developers are and provide efficient intuitive API's (plenty of MSFT API's are). Many developers currently use React for web. The V10 JS API is (at moment) terrible for React devs to get started with imho. I've wasted hours trying to get it working and all we need to do is upload an image to Azure Storage.

So - how do we, use this V10 library to get an image selected in the browser => up to Azure Storage?? And on this proviso that ```SharedKeyCredential``` is for some reason not supported in browsers.

A direct example would be helpful and I'm sure plenty others will benefit from it

**Comment by XiaoningLiu** [comment]
@SogoGolf Thanks for your feedback! Besides the SharedKeyCredential, what others blocked you from using storage V10 SDK with React?

We are working on improving storage JS SDKs, in the past few months, along with Azure SDK team, we did lots of improvements and changes to make JS V10 better. We call it track 2, and going to be preview released in next week. In Track2, developers can skip creating pipelines, and create a ServiceClient easily. See track 2 sample here: https://github.com/Azure/azure-sdk-for-js/tree/feature/storage/sdk/storage/storage-blob#code-samples

There are tons of commits just for the convenience improvements. https://github.com/Azure/azure-sdk-for-js/commits/feature/storage

**Comment by SogoGolf** [comment]
thanks @XiaoningLiu good to hear. I can't get past the SharedKeyCredential issue to use the V10 SDK at all, so for now that's the blocker.

from a quick review of the code examples the only other thing that seemed odd was
that ```pipeline``` setup & param. i took a look at the code sample you mentioned above (for Track2) but it still shows the use of ```pipeline``` ? I'll keep an eye out for the preview + perhaps drop a quick update here once its released, thanks.

**Comment by XiaoningLiu** [comment]
> thanks @XiaoningLiu good to hear. I can't get past the SharedKeyCredential issue to use the V10 SDK at all, so for now that's the blocker.
> 
> from a quick review of the code examples the only other thing that seemed odd was
> that `pipeline` setup & param. i took a look at the code sample you mentioned above (for Track2) but it still shows the use of `pipeline` ? I'll keep an eye out for the preview + perhaps if you can drop a quick update here once its released that would be great, thanks.

There are lots of PRs needs to be merged yet.

**Comment by glv1at118** [comment]
> > thanks @XiaoningLiu good to hear. I can't get past the SharedKeyCredential issue to use the V10 SDK at all, so for now that's the blocker.
> > from a quick review of the code examples the only other thing that seemed odd was
> > that `pipeline` setup & param. i took a look at the code sample you mentioned above (for Track2) but it still shows the use of `pipeline` ? I'll keep an eye out for the preview + perhaps if you can drop a quick update here once its released that would be great, thanks.
> 
> There are lots of PRs needs to be merged yet, like [Azure/azure-sdk-for-js#4058 (comment)](https://github.com/Azure/azure-sdk-for-js/pull/4058#discussion_r297347947)

Hello! Do you know this JS browser API is compatible with ionic framework on mobile environment or not? Right now I've been building an ionic mobile app and it will need to upload file to Azure Blob. Will the "createBlockBlobFromBrowserFile" ever work on the mobile environment? Any ideas?
